gcloud-java-dns/src/main/java/com/google/gcloud/testing/LocalDnsHelper.java

```diff
@@ -0,0 +1,1426 @@
+    } catch (IOException e) {
+      throw new IOException("Exception encountered when decoding request content.", e);
+    }
+  }
+
+  /**
+   * Generates a JSON response. Tested.
+   */
+  static Response toListResponse(List<String> serializedObjects, String pageToken,
+      boolean includePageToken) {
+    // start building response
+    StringBuilder responseBody = new StringBuilder();
+    responseBody.append("{\"projects\": [");
+    Joiner.on(",").appendTo(responseBody, serializedObjects);
+    responseBody.append("]");
+    // add page token only if exists and is asked for
+    if (pageToken != null && includePageToken) {
+      responseBody.append(",\"pageToken\": \"").append(pageToken).append("\"");
+    }
+    responseBody.append("}");
+    return new Response(HTTP_OK, responseBody.toString());
+  }
+
+  /**
+   * Prepares DNS records that are created by default for each zone.
+   */
+  private static ImmutableList<RrsetWrapper> defaultRecords(ManagedZone zone) {
+    ResourceRecordSet soa = new ResourceRecordSet();
+    soa.setTtl(21600);
+    soa.setName(zone.getDnsName());
+    soa.setRrdatas(ImmutableList.of(
+        // taken from the service
+        "ns-cloud-c1.googledomains.com. cloud-dns-hostmaster.google.com. 0 21600 3600 1209600 312"
+    ));
+    soa.setType("SOA");
+    ResourceRecordSet ns = new ResourceRecordSet();
+    ns.setTtl(21600);
+    ns.setName(zone.getDnsName());
+    ns.setRrdatas(zone.getNameServers());
+    ns.setType("NS");
+    RrsetWrapper nsWrapper = new RrsetWrapper(ns);
+    RrsetWrapper soaWrapper = new RrsetWrapper(soa);
+    ImmutableList<RrsetWrapper> results = ImmutableList.of(nsWrapper, soaWrapper);
+    nsWrapper.setId(getUniqueId(results));
+    soaWrapper.setId(getUniqueId(results));
+    return results;
+  }
+
+  /**
+   * Returns a list of four nameservers randomly chosen from the predefined set.
+   */
+  static List<String> randomNameservers() {
+    ArrayList<String> nameservers = Lists.newArrayList(
+        "dns1.googlecloud.com", "dns2.googlecloud.com", "dns3.googlecloud.com",
+        "dns4.googlecloud.com", "dns5.googlecloud.com", "dns6.googlecloud.com"
+    );
+    while (nameservers.size() != 4) {
+      int index = ID_GENERATOR.nextInt(nameservers.size());
+      nameservers.remove(index);
+    }
+    return nameservers;
+  }
+
+  /**
+   * Returns a hex string id (used for a dns record) unique withing the set of wrappers.
+   */
+  static String getUniqueId(List<RrsetWrapper> wrappers) {
+    TreeSet<String> ids = Sets.newTreeSet(Lists.transform(wrappers,
+        new Function<RrsetWrapper, String>() {
+          @Nullable
+          @Override
+          public String apply(@Nullable RrsetWrapper input) {
+            return input.id() == null ? "null" : input.id();
+          }
+        }));
+    String id;
+    do {
+      id = Long.toHexString(System.currentTimeMillis())
+          + Long.toHexString(Math.abs(ID_GENERATOR.nextLong()));
```

Comment:
Fine as is. [UUID.randomUUID()](https://docs.oracle.com/javase/7/docs/api/java/util/UUID.html#randomUUID()) could be used instead.